### PR TITLE
[NRK] Improve extractor

### LINKED ISSUE #3211: nrk.no -o "%(upload_date)s" returns NA
### Checklist

- [X] I'm reporting a site feature request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

I think the link is worldwide

### Example URLs

https://tv.nrk.no/serie/debatten/202202/NNFA51020322

### Description

using yt-dlp 2022.03.08.2

When using -o "%(upload_date)s" it returns NA

I'm not sure if this ever worked in yt-dlp - but it worked fine up until a couple of days ago, and I think it was when I switched my last script over from youtube-dl to yt-dlp.

I found nothing on nrk links in the bugtracker, but this seems to be the same: https://github.com/yt-dlp/yt-dlp/issues/1472

### Verbose log

```shell
yt-dlp -i https://tv.nrk.no/serie/debatten/202202/NNFA51020322 -o "%(series)s [%(upload_date)s] %(title)s.%(ext)s" -vU
[debug] Command-line config: ['-i', 'https://tv.nrk.no/serie/debatten/202202/NNFA51020322', '-o', '%(series)s/sesong_%(season_number)02d/%(series)s [%(upload_date)s] %(title)s-[ID]%(id)s[ID].%(ext)s', '-vU']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.2 [e491d06d3]
[debug] Python version 3.7.3 (CPython 64bit) - Linux-4.19.0-16-amd64-x86_64-with-debian-10.10
[debug] exe versions: ffmpeg 4.1.6-1, ffprobe 4.1.6-1
[debug] Optional libraries: brotli, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
Latest version: 2022.03.08.1, Current version: 2022.03.08.2
yt-dlp is up to date (2022.03.08.2)
[debug] [NRKTV] Extracting URL: https://tv.nrk.no/serie/debatten/202202/NNFA51020322
[debug] Using fake IP  (NO) as X-Forwarded-For
[debug] [NRK] Extracting URL: nrk:NNFA51020322
[NRK] NNFA51020322: Downloading manifest JSON
[NRK] NNFA51020322: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[NRK] NNFA51020322: Downloading metadata JSON
[NRK] NNFA51020322: Downloading programs JSON
[debug] Default format spec: bestvideo*+bestaudio/best
[info] NNFA51020322: Downloading 1 format(s): 2384
[debug] Invoking downloader on "https://nrk-od-74.akamaized.net/world/1376474/0/hls/nnfa51020322/1853524312-1203177498-prog_index.m3u8?version_hash=090cd527"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 269
[download] Destination: Debatten [NA] Debatten - 3. feb. · Er Rødt det nye folkepartiet.mp4
[download]   0.8% of ~737.55MiB at  7.94MiB/s ETA 01:14 (frag 2/269)^C
ERROR: Interrupted by user
```

### PR DESCRIPTION
**Pull request by hatienl0i261299**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Close #3211

### CODE REVIEW

File: yt_dlp/extractor/nrk.py [hunk 1]
```diff
@@ -247,6 +248,8 @@ def call_playback_api(item, query=None):
             'age_limit': age_limit,
             'formats': formats,
             'subtitles': subtitles,
+            'upload_date': unified_strdate(
+                str_or_none(try_get(manifest, lambda x: x['availability']['onDemand']['from'])))
```
Comment (pukkandan) [inline yt_dlp/extractor/nrk.py:252]:
```suggestion
                try_get(manifest, lambda x: x['availability']['onDemand']['from'], str))
```

**Comment by pukkandan** [comment]
Also, pls check https://github.com/yt-dlp/yt-dlp/issues/3211#issuecomment-1082861488

**Comment by hatienl0i261299** [comment]
> Also, pls check [#3211 (comment)](https://github.com/yt-dlp/yt-dlp/issues/3211#issuecomment-1082861488)

I checked, it seem like related to the utils function `unified_strdate`
image

**Comment by pukkandan** [comment]
If you have full timestamp available, why are you returning upload_date and not timestamp?

**Comment by hatienl0i261299** [comment]
> If you have full timestamp available, why are you returning upload_date and not timestamp?



> If you have full timestamp available, why are you returning upload_date and not timestamp?

Actually, API only return the update_time with format like this `2022-03-29T21:19:10.96+02:00`, it doesnt return timestamp.

**Comment by hatienl0i261299** [comment]
> If you have full timestamp available, why are you returning upload_date and not timestamp?

Actually, API only return the update_time with format like this 2022-03-29T21:19:10.96+02:00, it doesnt return timestamp.

**Comment by pukkandan** [comment]
that is a timestamp, not just a date 😕

**Pull request merged**